File: 1.1/src/c/cz/mg/vulkan/PFN_vkCmdDrawIndirectCountAMD.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkCmdDrawIndirectCountAMD_callNative(JNIEnv* env, jclass jc, jlong address, jlong commandBuffer, jlong buffer, jlong offset, jlong countBuffer, jlong countBufferOffset, jlong maxDrawCount, jlong stride){
    (void)env;
    (void)jc;
    PFN_vkCmdDrawIndirectCountAMD f = (PFN_vkCmdDrawIndirectCountAMD)jniLongToFunctionPointer(address);
    f(
        *((VkCommandBuffer*)jniLongToPointer(commandBuffer)),
        *((VkBuffer*)jniLongToPointer(buffer)),
        *((VkDeviceSize*)jniLongToPointer(offset)),
        *((VkBuffer*)jniLongToPointer(countBuffer)),
        *((VkDeviceSize*)jniLongToPointer(countBufferOffset)),
        *((uint32_t*)jniLongToPointer(maxDrawCount)),
        *((uint32_t*)jniLongToPointer(stride))
    );
}


void Java_cz_mg_vulkan_PFNvkCmdDrawIndirectCountAMD_callSimplifiedNative(JNIEnv* env, jclass jc, jlong address, jlong commandBuffer, jlong buffer, jlong offset, jlong countBuffer, jlong countBufferOffset, jint maxDrawCount, jint stride){
    (void)env;
    (void)jc;
    PFN_vkCmdDrawIndirectCountAMD f = (PFN_vkCmdDrawIndirectCountAMD)jniLongToFunctionPointer(address);
    f(
        *((VkCommandBuffer*)jniLongToPointer(commandBuffer)),
        *((VkBuffer*)jniLongToPointer(buffer)),
        (VkDeviceSize)offset,
        *((VkBuffer*)jniLongToPointer(countBuffer)),
        (VkDeviceSize)countBufferOffset,
        (uint32_t)maxDrawCount,
        (uint32_t)stride
    );
}

File: 1.1/src/cz/mg/vulkan/PFNvkCmdDrawIndirectCountAMD.java
package cz.mg.vulkan;

public class PFNvkCmdDrawIndirectCountAMD extends VkFunctionPointer {
    public PFNvkCmdDrawIndirectCountAMD() {
    }

    protected PFNvkCmdDrawIndirectCountAMD(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkCmdDrawIndirectCountAMD(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkCmdDrawIndirectCountAMD(long value) {
        setValue(value);
    }

    public PFNvkCmdDrawIndirectCountAMD(VkInstance instance) {
        super(instance, new VkString("vkCmdDrawIndirectCountAMD"));
    }

    public void call(VkCommandBuffer commandBuffer, VkBuffer buffer, long offset, VkBuffer countBuffer, long countBufferOffset, int maxDrawCount, int stride){
        callNative(getValue(), commandBuffer != null ? commandBuffer.getVkAddress() : VkPointer.getNullAddressNative(), buffer != null ? buffer.getVkAddress() : VkPointer.getNullAddressNative(), offset, countBuffer != null ? countBuffer.getVkAddress() : VkPointer.getNullAddressNative(), countBufferOffset, maxDrawCount, stride);
    }

    protected static native void callNative(long vkaddress, long commandBuffer, long buffer, long offset, long countBuffer, long countBufferOffset, int maxDrawCount, int stride);

}
